engine/src/main/java/org/terasology/rendering/logic/SkeletonRenderer.java

```diff
@@ -288,20 +288,16 @@ public void renderOpaque() {
                 if (boneLocation != null) {
                     Matrix4f boneTransform = new Matrix4f();
                     boneLocation.getRelativeTransform(boneTransform, entity);
-                    boneTransform.mul(JomlUtil.from(bone.getInverseBindMatrix()).transpose());
-                    boneTransforms[bone.getIndex()] = boneTransform;
+                    boneTransform.mul(bone.getInverseBindMatrix());
+                    boneTransforms[bone.getIndex()] = boneTransform.transpose();
                 } else {
                     logger.warn("Unable to resolve bone \"{}\"", bone.getName());
                     boneTransforms[bone.getIndex()] = new Matrix4f();
 
                 }
             }
             ((OpenGLSkeletalMesh) skeletalMesh.mesh).setScaleTranslate(skeletalMesh.scale, skeletalMesh.translate);
-            ((OpenGLSkeletalMesh) skeletalMesh.mesh).render(Arrays.stream(boneTransforms).map(k -> {
-                org.terasology.math.geom.Matrix4f t = JomlUtil.from(k);
-                t.transpose();
-                return t;
-            }).collect(Collectors.toList()));
+            ((OpenGLSkeletalMesh) skeletalMesh.mesh).render(Arrays.stream(boneTransforms).collect(Collectors.toList()));
```

Comment:
No need to use a Stream here:
```suggestion
            ((OpenGLSkeletalMesh) skeletalMesh.mesh).render(Arrays.asList(boneTransforms));
```